control-plane/api/v1alpha1/gatewaypolicy_types.go

```diff
@@ -115,3 +115,21 @@ type GatewayJWTClaimVerification struct {
 	//   that this value matches.
 	Value string `json:"value"`
 }
+
+// GatewayPolicyStatus defines the observed state of the gateway
+type GatewayPolicyStatus struct {
```

Comment:
we needed a specific status type in order to add the status, similar to how the `Gateway` type in the k8s spec has a specific type for `GatewayStatus`